### ISSUE TITLE
Inconsistent results displayed in the store

### ISSUE THREAD

**Issue by bobeal** [ozwillo/ozwillo-portal#407]
When searching for "Open Data", 5 Open Data applications or services are displayed in the search results :

![screenshot from 2017-11-14 11-49-09](https://user-images.githubusercontent.com/175382/32776172-fd7be53a-c931-11e7-9461-3ace69f0fea8.png)

When scrolling through all the results, only 2 results are displayed for Open Data applications or services :

![screenshot from 2017-11-14 11-48-11](https://user-images.githubusercontent.com/175382/32776249-3bdf5000-c932-11e7-8053-4bdfcbaa7983.png)

**Comment by tbroyer** [comment]
This is actually a bug in the Kernel!

Fwiw, when using search query terms, an Elastic index is used, whereas otherwise the list comes right from MongoDB; see https://github.com/ozwillo/ozwillo-kernel/blob/b05caba044231f3bc8fa6b3300305ef6058ba41e/oasis-webapp/src/main/java/oasis/catalog/RoutingCatalogEntryRepository.java

There are 2 issues actually:
 * it appears that some hidden services are indexed in Elastic; I'd like to blame a "broken pipe" at some point between the Kernel and Elastic which has lead to the index being out of date (with the services having been indexed while visible, and not properly un-indexed when changed to hidden). I'll re-index the catalog, it should solve the problem. We need to setup some monitoring to make sure it won't happen again though, and investigate to understand how (and when) it happened (and possibly a cron job to re-index everything on a regular basis; we'd an "smart" re-indexing strategy though, as currently we need to delete the index and re-index from scratch, so there's a small time-window where the index is empty or incomplete).
 * when querying MongoDB (i.e. when not using query terms, such as when paging through the whole catalog), some visible services can actually not be shown. This is a regression due to the change from `visible`/`restricted` properties to `visibility`/`access_control` ones; where the query explicitly filters for `visible: true` services (where some might nowadays no longer have that property but have `visibility: 'VISIBLE'`). A short-term solution could be to set `visible:true` whenever `visibility: 'VISIBLE', access_control: 'ANYONE'` (I would have expected this to be the case actually, and we have a unit test that covers this, at least with Jackson, though not Jongo). The long-term solution will need a fix in the code though.

You can compare the results with something like:
```shell
curl 'https://kernel.ozwillo-preprod.eu/m/search?hl=fr&q=open+data' |jq 'map({"id":.id,"type":.type,"name":.name})'
```
(tweak or remove the `q` query-string parameter, also play with the `start` and `limit` parameters for pagination –default to `start=0&limit=25`)

**Comment by tbroyer** [comment]
OK, so both were actually bugs in the code. Fixed in ozwillo/ozwillo-kernel@57ffccb9e0b8f0ccaf15bb2ac747005315c11cd5 and ozwillo/ozwillo-kernel@734560d875b6ba78ffd826b21561e27cf9bc3b02;released as v1.48.2 and deployed in preprod.
